contracts/swap/LatteSwapOptimalDeposit.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+
+import "./interfaces/ILatteSwapFactory.sol";
+import "./interfaces/ILatteSwapPair.sol";
+import "./interfaces/ILatteSwapRouter.sol";
+import './interfaces/IWBNB.sol';
+
+import "./libraries/LatteSwapMath.sol";
+import "./libraries/SafeToken.sol";
+
+contract LatteSwapOptimalDeposit is ReentrancyGuardUpgradeable {
+  using SafeToken for address;
+  using SafeMathUpgradeable for uint256;
+
+  ILatteSwapFactory public factory;
+  ILatteSwapRouter public router;
+  address public WBNB;
+
+  /// @dev Create a new add two-side optimal strategy instance.
+  /// @param _router The Router smart contract.
+  function initialize(ILatteSwapRouter _router, address _WBNB) external initializer {
+    ReentrancyGuardUpgradeable.__ReentrancyGuard_init();
+
+    factory = ILatteSwapFactory(_router.factory());
+    router = _router;
+    WBNB = _WBNB;
+  }
+
+  /// @dev Compute swap amount to optimal deposit
+  /// @param amtA amount of tokenA desired to deposit
+  /// @param amtB amonut of tokenB desired to deposit
+  /// @param resA amount of tokenA in reserve
+  /// @param resB amount of tokenB in reserve
+  function calSwapAmount(
+    uint256 amtA,
+    uint256 amtB,
+    uint256 resA,
+    uint256 resB
+  ) internal pure returns (uint256 swapAmt, bool isReversed) {
+    if (amtA.mul(resB) >= amtB.mul(resA)) {
+      swapAmt = _calSwapAmount(amtA, amtB, resA, resB);
+      isReversed = false;
+    } else {
+      swapAmt = _calSwapAmount(amtB, amtA, resB, resA);
+      isReversed = true;
+    }
+  }
+
+  /// @dev Compute swap amount helper
+  /// @param amtA amount of tokenA desired to deposit
+  /// @param amtB amonut of tokenB desired to deposit
+  /// @param resA amount of tokenA in reserve
+  /// @param resB amount of tokenB in reserve
+  function _calSwapAmount(
+    uint256 amtA,
+    uint256 amtB,
+    uint256 resA,
+    uint256 resB
+  ) internal pure returns (uint256) {
+    require(amtA.mul(resB) >= amtB.mul(resA), "Reversed");
+
+    uint256 a = 998;
```

Comment:
incorrect fee, we are using 0.25 as a fee, thus change it to 9975

Reply:
updated